cmake/Sanitizers.cmake

```diff
@@ -0,0 +1,9 @@
+function(enable_sanitizers project_name)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    option(ENABLE_COVERAGE "Enable coverage report for clang" OFF)
+    if (ENABLE_COVERAGE)
```

Comment:
`message(STATUS ...` just to be consistent with `BUILD_TESTING`, etc?

Reply:
Definitely should add a message